app/src/pages/inside/testsPage/historyLine/historyLine.jsx

```diff
@@ -0,0 +1,170 @@
+      method: 'get',
+    })
+      .then((response) => {
+        this.getHistoryItemsToRender(response.reverse());
+      })
+      .catch(() => {
+        this.props.showNotification({
+          message: this.props.intl.formatMessage(messages.loadHistoryErrorNotification),
+          type: NOTIFICATION_TYPES.ERROR,
+        });
+      });
+  };
+
+  getHistoryItemProps = (filteredLaunchHistoryItem) => {
+    const itemProps = filteredLaunchHistoryItem[0];
+    let itemStatus = '';
+
+    if (!filteredLaunchHistoryItem.length) {
+      itemStatus = NOT_FOUND.toUpperCase();
+    } else if (filteredLaunchHistoryItem.length > 1) {
+      itemStatus = MANY.toUpperCase();
+      delete itemProps.issue;
+      delete itemProps.statistics;
+    } else {
+      itemStatus = itemProps.status;
+    }
+    return {
+      ...itemProps,
+      status: itemStatus,
+    };
+  };
+
+  getHistoryItemsToRender = (itemsHistory) => {
+    const currentLaunch = itemsHistory.pop();
+    const currentLaunchItem = currentLaunch.resources.find((item) => item.id === itemExample);
+
+    const historyItems = itemsHistory.map((historyItem) => {
+      const filteredSameHistoryItems = historyItem.resources.filter(
+        (item) => item.uniqueId === currentLaunchItem.uniqueId,
+      );
+
+      return {
+        ...this.getHistoryItemProps(filteredSameHistoryItems),
+        launchNumber: historyItem.launchNumber,
+      };
+    });
+
+    currentLaunchItem.launchNumber = currentLaunch.launchNumber;
+    historyItems.push(currentLaunchItem);
+
+    this.setState({
+      itemsToRender: this.calculateGrowthDuration(historyItems.reverse()),
+      selectedLaunch: currentLaunch.launchNumber,
+    });
+  };
+
+  calculateGrowthDuration = (historyItems) => {
+    const historyItemsLastIndex = historyItems.length - 1;
+
+    historyItems.forEach((item, index) => {
+      const newItemData = item;
+
+      if (index < historyItemsLastIndex && this.validForDurationGrowth(item)) {
+        let prevItemIndex = index + 1;
+        while (
+          !this.validForDurationGrowth(historyItems[prevItemIndex]) &&
+          prevItemIndex < historyItemsLastIndex
+        ) {
+          prevItemIndex += 1;
+        }
+
+        if (prevItemIndex <= historyItemsLastIndex) {
+          const prevDuration =
+            historyItems[prevItemIndex].end_time - historyItems[prevItemIndex].start_time;
+          const currentDuration = item.end_time - item.start_time;
+          const growth = currentDuration / prevDuration - 1;
+          growth > 0 && (newItemData.growthDuration = `+${Math.round(growth * 100)}%`);
+        }
+      }
+
+      return newItemData;
+    });
+
+    return historyItems.reverse();
+  };
+
+  validForDurationGrowth = (item) =>
+    item.status === FAILED.toUpperCase() || item.status === PASSED.toUpperCase();
+
+  render() {
+    return (
+      <div className={cx('history-line')}>
+        {this.state.itemsToRender && (
+          <HistoryLineToggle
```

Comment:
You don't need one more component to render items. Just do it right here.

Reply:
Fixed